src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -3647,10 +3644,8 @@ static int io_callback(sd_event_source *s, int fd, uint32_t revents, void *userd
         /* Note that this is called both on input_fd, output_fd, as well as inotify_fd events */
 
         r = sd_bus_process(bus, NULL);
-        if (r < 0) {
-                log_debug_errno(r, "Processing of bus failed, closing down: %m");
-                bus_enter_closing(bus);
-        }
+        if (r < 0)
+                bus_enter_closing(bus, log_debug_errno(r, "Processing of bus failed, closing down: %m"));
```

Comment:
FWIW, I liked the previous form better. Chaining `log_debug_errno` like this seems a bit too much.